src/Converter/ShipmentLineItemsConverter.php

```diff
@@ -43,14 +51,21 @@ private function convertUnitRefundToLineItem(UnitRefundInterface $unitRefund): L
         Assert::notNull($shippingAdjustment);
         Assert::lessThanEq($unitRefund->total(), $shippingAdjustment->getAmount());
 
+        /** @var ShipmentInterface $shipment */
+        $shipment = $this->shipmentRepository->find(['id' => $unitRefund->id()]);
+        Assert::notNull($shipment);
+        /** @var AdjustmentInterface $adjustment */
+        $adjustment = $shipment->getUnits()->first()->getAdjustments()->first();
```

Comment:
Are you sure that the first adjustment of the first unit of a shipment always contains information about shipping costs?